assets/js/hooks/index.ts

```diff
@@ -8,6 +8,32 @@ import { initiateSaveAndRun } from '../common';
 
 export { LogLineHighlight, ElapsedIndicator, TabSelector };
 
+export const EditMandatoryScope = {
+  mounted() {
+    this.el.addEventListener('dblclick', e => {
+      const scopeValue = this.el.dataset.scope;
+      this.pushEventTo(this.el, 'edit_mandatory_scope', { scope: scopeValue });
+    });
+  },
+};
+
+export const EditOptionalScope = {
+  mounted() {
+    this.el.addEventListener('dblclick', e => {
+      const scopeValue = this.el.dataset.scope;
+      this.pushEventTo(this.el, 'edit_optional_scope', { scope: scopeValue });
+    });
+  },
+};
```

Comment:
These two can be made into one, if you supply like a `phx-event` (can be anything you want) and maybe `phx-value-scope`. While those are 'liveview' hook conventions (and we are free to do whatever), this could be made into. a `DoubleClick` hook.

Reply:
This is done. Thanks